src/classes/OpportunityTriggerHelper.cls

```diff
@@ -0,0 +1,421 @@
+                                }
+                            }
+                            else {
+                                for (Database.Error error : result.getErrors()) {
+                                    message2 = error.getStatusCode() + ': ' + error.getMessage();
+                                }
+                            }
+                        }
+                    }
+                    else if (sensorName.substring(1,3).isNumeric()) {
+                        sensorType = 'Automobile Tag';
+                        if (flag == true) {
+                            if (result.isSuccess()) {
+                                message1 = sensorType + ' Inserted Under Installation Account : ' + (String)recordName.get('Account__c');
+                            }
+                            else {
+                                for (Database.Error error : result.getErrors()) {
+                                    message2 = error.getStatusCode() + ': ' + error.getMessage();
+                                }
+                            }
+                        }
+                        else {
+                            if (result.isSuccess()) {
+                                if (opp.AccountId == (String)recordName.get('Account__c')) {
+                                    message1 = sensorType + ' is moved from Shipment Account ' + opp.Shipment_Account__c + ' to Installation Account ' + opp.AccountId;                                    
+                                }
+                                else {
+                                    message1 = sensorType + ' is moved from Installation Account ' + opp.AccountId + ' to Shipment Account ' + opp.Shipment_Account__c;                                   
+                                }
+                            }
+                            else {
+                                for (Database.Error error : result.getErrors()) {
+                                    message2 = error.getStatusCode() + ': ' + error.getMessage();
+                                }
+                            }
+                        }
+                    }
+                }
+
+                // Add message to HTML Body
+                htmlBody1 += '<tr><td>' + sensorType + '</td><td>' + sensorName + '</td><td>' + message1 + '</td></tr>';                        
+                htmlBody2 += '<tr><td>' + sensorType + '</td><td>' + sensorName + '</td><td>' + message2 + '</td></tr>';   
+                message2Reference = message2;                     
+            }
+
+            // Close HTML Body
+            htmlBody1 += '</table><br>';
+            htmlBody2 += '</table><br>';
+            if (message2Reference != null) {
+                emailMsg.setHtmlBody(htmlBody1 + htmlBody2);                
```

Comment:
WHy are you adding it here ?

Reply:
I am adding bodies here Because htmlBody1 is for Successful INSERT, UPDATE Operations and htmlBody2 is for Unsuccessful Operations.
But only Duplicate error is added, Other errors like Invalid Record is not added in htmlBody2. It shows error while create opportunity and allow to create Opportunity with Invalid Record.
Means that only duplicate error is added in htmlBody2 Invalid Record error is not added in the table.